### PR TITLE
Add combineSpectra,MSnExp method (issue #474)

### PR DESCRIPTION
**Pull request by jorainer**
- Add the combineSpectra,MSnExp method.
- Add related documentation and unit tests.
- Remove warnings in meanMzInts and consensusSpectrum to avoid confusions for
  general input data (such as MS2 spectra from DIA data).

This fixes issue #474.

### CODE REVIEW

File: R/functions-Spectrum.R [hunk 1]
```diff
@@ -979,10 +981,13 @@ meanMzInts <- function(x, ..., intensityFun = base::mean, weighted = FALSE,
         mz_groups <- .group_mz_values(sqrt(mzs), mzd = mzd)
     else
         mz_groups <- .group_mz_values(mzs, mzd = mzd, ppm = ppm)
-    if (length(unique(mz_groups)) < length(x[[main]]@mz))
-        warning("Got less m/z groups than m/z values in the original spectrum.",
-                " Most likely the data is not profile-mode LCMS data or ",
-                "'mzd' is too large.")
+    ## Disable the warning below, as this does not apply to all use cases: for
```
Comment (sgibb) [inline R/functions-Spectrum.R]: Why not remove the whole commented block?

File: R/functions-Spectrum.R [hunk 2]
```diff
@@ -1220,9 +1227,12 @@ consensusSpectrum <- function(x, mzd, minProp = 0.5, intensityFun = base::max,
     mzs <- mzs[keep]
     ints <- ints[keep]
     mz_groups <- .group_mz_values(mzs, mzd = mzd, ppm = ppm)
-    if (length(unique(mz_groups)) < sum(xnew@intensity > 0))
-        warning("Got less m/z groups than m/z groups in the first spectrum. ",
-                "Most likely `mzd` is too large.")
+    ## Disable the warning below, as this does not apply to all use cases: for
```
Comment (sgibb) [inline R/functions-Spectrum.R]: Same as above.

File: R/functions-Spectrum.R (hunk 1 above)
Comment (jorainer) [reply R/functions-Spectrum.R]: Done

File: R/functions-Spectrum.R (hunk 2 above)
Comment (jorainer) [reply R/functions-Spectrum.R]: Done

**Comment by jorainer** [comment]
Just a reminder for @lgatto: please push this to Bioconductor at some point.

**Pull request merged**